embed/etcd.go

```diff
@@ -400,6 +408,29 @@ func (e *Etcd) serve() (err error) {
 			e.errHandler(s.serve(e.Server, &e.cfg.ClientTLSInfo, h, e.errHandler))
 		}(sctx)
 	}
+
+	if len(e.cfg.ListenMetricsUrls) > 0 {
+		// TODO: maybe etcdhttp.MetricsPath or get the path from the user-provided URL
+		metricsMux := http.NewServeMux()
+		metricsMux.Handle("/metrics", prometheus.Handler())
+
+		for _, murl := range e.cfg.ListenMetricsUrls {
+			if murl.Scheme == "https" {
```

Comment:
why can't this work like the other listen urls where if http://... gets plaintext, https:// gets tls, unix:// gets unix socket, etc? seems like the listener scheme handling/semantics should be the same if possible to avoid confusion